common/modules/locking_module/src/lock_with_energy_module.rs

```diff
@@ -36,6 +36,46 @@ pub trait LockWithEnergyModule {
             .execute_on_dest_context()
     }
 
+    #[inline]
+    fn lock_tokens(
+        &self,
+        token_id: EgldOrEsdtTokenIdentifier,
+        amount: BigUint,
+        lock_epochs: OptionalValue<u64>,
+    ) -> EgldOrEsdtTokenPayment<Self::Api> {
+        self.lock_common(OptionalValue::None, token_id, amount, lock_epochs)
+    }
+
+    #[inline]
+    fn lock_tokens_and_forward(
+        &self,
+        to: ManagedAddress,
+        token_id: EgldOrEsdtTokenIdentifier,
+        amount: BigUint,
+        lock_epochs: OptionalValue<u64>,
+    ) -> EgldOrEsdtTokenPayment<Self::Api> {
+        self.lock_common(OptionalValue::Some(to), token_id, amount, lock_epochs)
+    }
+
+    fn lock_common(
+        &self,
+        opt_dest: OptionalValue<ManagedAddress>,
+        token_id: EgldOrEsdtTokenIdentifier,
+        amount: BigUint,
+        lock_epochs: OptionalValue<u64>,
+    ) -> EgldOrEsdtTokenPayment<Self::Api> {
+        let unlock_epoch = match lock_epochs {
+            OptionalValue::Some(epochs) => epochs,
+            OptionalValue::None => self.lock_epochs().get(),
+        };
+        let mut proxy_instance = self.get_locking_sc_proxy_instance();
+
+        proxy_instance
+            .lock_tokens_endpoint(unlock_epoch, opt_dest)
```

Comment:
```suggestion
        opt_lock_epochs: Option<u64>,
    ) -> EgldOrEsdtTokenPayment<Self::Api> {
        let lock_epochs = match lock_epochs {
            Some(epochs) => epochs,
           None => self.lock_epochs().get(),
        };
        let mut proxy_instance = self.get_locking_sc_proxy_instance();

        proxy_instance
            .lock_tokens_endpoint(lock_epochs, opt_dest)
```

Reply:
Also modify to use `Option` instead of `OptionalValue` wherever this function is used.